java-client/src/test/java/org/opensearch/client/opensearch/experiments/ParsingTests.java

```diff
@@ -85,4 +86,44 @@ public void testFoo() throws Exception {
       throw new JsonParsingException(je.getMessage() + " at " + je.getLocation(), je, je.getLocation());
     }
   }
+
+  @Test
+  public void testIndexSettingsMappingParsing() {
```

Comment:
Can we also add a test for index settings mapping during index creation ([here](https://github.com/opensearch-project/opensearch-java/blob/main/java-client/src/test/java/org/opensearch/client/opensearch/integTest/AbstractRequestIT.java#L114))?